Token migration fixups

starbelly opened the pull request:
# Changes

 - CamelCase namespace for migration module
   * Since namespace was Guardian.DB.Gen.Migration and not
   Guardian.Db.Gen.Migration mix raised an error in at least <= 1.5.2
 - Add test case for guardian.db.gen.migration
 - Removed priv/test from repo in favor of setting up per test case
 - Updated .gitignore per removing priv/test
 - Added "aud" parameter to tokens created in "purge stale tokens" tests
 - Add test/support/file_helpers.ex with tmp_path/1

# Notes 
 - It should be noted that there is probably a better way to do test setup. I had to adjust to running the migrations on demand, as  this seemed like the path of least resistance. Running migrate in the case like this seems sub-optimal, because if you want to do async test then you're going to have problems. It may be optimal to refactor the task test to use a different Repo. But before doing that, figured a review of what's been done thus far is in order.

- It should also be noted that per removing priv/test, two tests ended up in failure per not passing aud parameter. It seemed like the tests and the static test migration were out of sync with what's currently generated.

yordis commented on mix.exs:
```diff
@@ -66,6 +66,6 @@ defmodule Guardian.DB.Mixfile do
 
   defp aliases do
-    [test: ["ecto.create --quiet", "ecto.migrate", "test"]]
+    [test: ["ecto.drop --quiet", "ecto.create --quiet", "test"]]
```
- Why do you want to drop and then create the database every time?
- Why did you remove the migration?

yordis commented on priv/test/migrations/20160929125415_migrations.exs:
```diff
@@ -1,22 +0,0 @@
-defmodule Guardian.DB.Test.Repo.Migrations do
```
Why do you remove this file?

starbelly replied:
This was per needing to wanting to auto-create tables and migrations. But upon refactoring to use different repo, I will re-add this. However, removing it did result in failures as pointed out in my notes.

yordis commented on test/mix/tasks/guardian.db.gen.migration_test.exs:
```diff
@@ -0,0 +1,21 @@
+defmodule Mix.Tasks.Guardian.Db.Gen.MigrationTest do
```
When do I need to call this task?

hassox replied:
This is a test mate. It's testing that the migration runs correctly and doesn't bomb out.

yordis commented on test/support/data_case.ex:
```diff
@@ -10,6 +11,15 @@ defmodule Guardian.DB.Test.DataCase do
+  setup_all do
+    tmp_path = tmp_path("priv/test/migrations")
+    File.mkdir_p!(tmp_path)
+    Mix.Task.run("guardian.db.gen.migration")
```
Why would I want te calling this task every single time the setup happens and the migrate?

The only time you would need to migrate is that you are changing the schema of the `Token` which wouldn't happen that often at all. 

So basically you will need to do `MIX_ENV=test mix ecto.reset` for be able to reset your database but I wouldn't add time to the testing.

Could you please put some benchmark on this?

starbelly replied:
As stated, initial PR. I think it's going to be better to use a stub Repo as Ecto does in their migration tests. I will refactor.

hassox replied:
@yordis I believe the intention is to run against the generator rather than a hard coded test migration file. Please correct me if I'm wrong @starbelly.

The setup_all runs only once for the case which we should be able to limit to a single test file so it should only run once.

starbelly replied:
That is the itention, but as stated, I think it's going to be better to to set the test up how ecto does in their gen migration tests: 

```elixir
defmodule Repo do
    def __adapter__ do
      true
    end

    def config do
      [priv: "tmp/#{inspect(Ecto.Gen.Migration)}", otp_app: :ecto]
    end
  end
```

yordis commented:
@starbelly let me know when this is finished so I remove the WIP, I will leave it until we feel confident to bring all the team to review it before it gets merge.

hassox commented:
This seems essentially equivalent with what is there. I'm assuming that as you move forward only the db adapter will actually use the db test so there won't be a problem with the migration part. Is that correct?

ghost commented:
That is correct. However, need to look into something first before going further. Why is the migration being executed? Can we stop that from happening? That's the optimal solution, but not seeing how to prevent that. I do not believe this was the behavior I saw in ecto tests, but will have to go back and look. 
 In addition, I don't think we should put the static files back. As noted removing the static files uncovered a bug in two tests (they were out of sync with what is currently generated). Thus, two places to do this is in test_helper.exs or per case.

Commit:
Token migration fixups

 - CamelCase namespace for migration module
   * Since namespace was Guardian.DB.Gen.Migration and not
   Guardian.Db.Gen.Migration mix raised an error in at least <= 1.5.2
 - Add test case for guardian.db.gen.migration
 - Changed test alias to always gen migration before executing ecto.migrate
 - Removed priv/test from repo in favor of setting up per test case
 - Updated .gitignore per removing priv/test
 - added aud paramter to token's created in "purge stale tokens" tests
 - Add test/support/file_helpers.ex

ghost commented:
@hassox @yordis This iteration is better over all and less invasive. So, we use the stub repo in the task test but pointing to a different tmp dir. Moved ecto.migrate call back to mix alias for test and added gen migration task in that list.

Commit:
Update README.md per migration task rename

Commit:
don't use abbreviations in error message

Commit:
Fixups

Commit:
remove /tmp from gitignore

fixup

Commit:
remove extra blank lines

hassox commented:
@yordis I would like to approve this to unblock @starbelly. Is there anything outstanding that needs to be done?

yordis commented:
@starbelly could you squash the commit please?

hassox commented:
@yordis I'll squash it in the merge
